src/bid/encoding.rs

```diff
@@ -9,58 +9,66 @@
 //! See: https://hackmd.io/@7dpNYqjKQGeYC7wMlPxHtQ/BkfS78Y9L
 
 use super::Bid;
+use dusk_bls12_381::BlsScalar;
+#[cfg(feature = "std")]
 use dusk_plonk::constraint_system::ecc::Point as PlonkPoint;
+#[cfg(feature = "std")]
 use dusk_plonk::prelude::*;
-use poseidon252::{sponge::sponge::*, StorageScalar};
+use poseidon252::sponge::sponge::*;
 
 // 1. Generate the type_fields Scalar Id:
 // Type 1 will be BlsScalar
 // Type 2 will be JubJubScalar
 // Type 3 will be JubJubAffine coordinates tuple
 // Type 4 will be u32
 // Type 5 will be PoseidonCipher
+// Type 6 will be u64
 // Byte-types are treated in Little Endian.
 // The purpose of this set of flags is to avoid collision between different
 // structures
-const TYPE_FIELDS: [u8; 32] = *b"53313110000000000000000000000000";
+const TYPE_FIELDS: [u8; 32] = *b"53313116000000000000000000000000";
 
 impl Bid {
     /// Calculate the one-way BlsScalar representation of the Bid
     pub fn hash(&self) -> BlsScalar {
         // Generate an empty vector of `Scalar` which will store the
         // representation of all of the `Bid` elements.
-        let mut words_deposit = Vec::new();
+        let mut words_deposit = [BlsScalar::zero(); 13];
```

Comment:
We should have two functions: `hash_inputs` that will output the hash inputs of the `Bid` structure, suck as `fn hash_inputs(&self) -> [BlsScalar; 13]`. The second one is the `hash` that will `fn hash(&self) -> BlsScalar` and use internally the `hash_inputs`

This is due to the fact that we may need to perform the hash outside the library with the `PoseidonLeaf` trait contract implementation

Reply:
Agree. We discussed that in the past I remember. Fixing it.